providers/sparkpost/src/lib/sparkpost.provider.ts

```diff
@@ -0,0 +1,45 @@
+import {
+  ChannelTypeEnum,
+  IEmailOptions,
+  IEmailProvider,
+  ISendMessageSuccessResponse,
+} from '@notifire/core';
+
+import SparkPost from 'sparkpost';
+export class SparkpostEmailProvider implements IEmailProvider {
+  id = 'sparkpost';
+  channelType = ChannelTypeEnum.EMAIL as ChannelTypeEnum.EMAIL;
+
+  private sparkpostClient: SparkPost;
+  constructor(
+    private config: {
+      apiKey: string;
+    }
+  ) {
+this.sparkpostClient = SparkPost(config.apiKey);
+  }
+
+  async sendMessage(
+    options: IEmailOptions
+  ): Promise<ISendMessageSuccessResponse> {
+
+const response= this.sparkpostClient.transmissions.send({
+  options: {
+    sandbox: true
+  },
+  content: {
+    from: options.from,
+    subject: options.subject,
+    html:options.html
+  },
+  recipients: [
+    {address: options.to}
```

Comment:
I think we have support for multiple `to` subjects, maybe it's better to check if it's an array or string and do `map` if needed here.